sci-libs/primegen/primegen-0.97.recipe

```diff
@@ -0,0 +1,78 @@
+SUMMARY="Fast C/C++ prime number generator library"
+DESCRIPTION="primegen is a small, fast library to generate prime numbers \
+in order. It generates the 50847534 primes up to 1000000000 in just 8 \
+seconds on a Pentium II-350; it prints them in decimal in just 35 seconds. \
+primegen can generate primes up to 1000000000000000, although it is not \
+optimized for primes past 32 bits. It uses the Sieve of Atkin instead of the \
+traditional Sieve of Eratosthenes."
+HOMEPAGE="https://cr.yp.to/primegen.html"
+COPYRIGHT="1999, D. J. Bernstein"
+LICENSE="Public Domain"
+REVISION="1"
+SOURCE_URI="https://cr.yp.to/primegen/primegen-$portVersion.tar.gz"
+CHECKSUM_SHA256="54285baf8eed9e421ff2220a2112d38cfb20c1ebef6014ef3f0004c22c95f40d"
+PATCHES="primegen-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+
+PROVIDES="
+	primegen = $portVersion
+	cmd:primegaps = $portVersion
+	cmd:primes = $portVersion
+	cmd:primespeed = $portVersion
+	lib:libprimegen
+	"
+REQUIRES="
+	haiku
+	"
+
+PROVIDES_devel="
+	primegen_devel = $portVersion
+	devel:libprimegen
+	"
+REQUIRES_devel="
+	primegen == $portVersion base
+	"
+
+BUILD_PREREQUIRES="
+	cmd:make
+	cmd:gcc
+	cmd:nroff
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+
+BUILD()
+{
+	make
+}
+
+INSTALL()
+{
+	install -d $binDir $docDir $manDir/man1 $manDir/man3
+	install -d $includeDir/primegen $developLibDir $libDir
+	install -t $binDir primegaps primes primespeed
+	install -t $docDir README
+	install -t $manDir/man1 primegaps.1 primes.1 primespeed.1
+	install -t $manDir/man3 primegen.3
+	install -t $includeDir/primegen primegen.h uint32.h uint64.h
+	install -T primegen.a $developLibDir/libprimegen.a
+	install -T primegen.so $libDir/libprimegen.so
+	prepareInstalledDevelLib libprimegen
+	fixPkgconfig
+
+	# devel package
+	packageEntries devel \
+		$developDir \
+		$manDir/man3
+}
+
+TEST()
+{
+	# 32-bit test
+	[ "$(primes 1 1000000 | md5sum)" != "c13929ee9d2aea8f83aa076236079e94  -" ] && exit 1
+	# all tests passed
+	exit 0
```

Comment:
I agree that we *currently* don't need to replace `primes` by `./primes`, but we can't be 100% sure it will always work without the `./`, so I'd really recomment replacing `primes` by `./primes`.
And while at it, I'd even do a few more cosmetic changes to have either:
```
[ "$(./primes 1 1000000 | md5sum)" != "c13929ee9d2aea8f83aa076236079e94  -" ]
```
or
```
[ "$(./primes 1 1000000 | md5sum | cut -d" " -f1 )" != c13929ee9d2aea8f83aa076236079e94 ]
```
The reason why I would prefer the second solution is that we don't need to care about what md5sum outputs right after the checksum. I know this is probably never going to change, but this way we don't hardcode an arbitrary strng (the two spaces and the dash) in our test.

The other advantage of removing the `&& exit 0` (and using `=` instead of `!=` is that we can use the exit code from the `[` command (which, as you know, is a kind of synonym for the `test` command) and, hence, have a TEST function with just one line in its body.

Personally, I'd use
```
test "$(./primes 1 1000000 | md5sum | cut -d" " -f1 )" != c13929ee9d2aea8f83aa076236079e94
```
but I'm perfectly OK with
```
[ "$(./primes 1 1000000 | md5sum | cut -d" " -f1 )" != c13929ee9d2aea8f83aa076236079e94 ]
```
too.
Well, I hope I'm not being too picky ;-)
So if you change TEST to:
```
TEST()
{
	[ "$(./primes 1 1000000 | md5sum)" != "c13929ee9d2aea8f83aa076236079e94  -" ]
]
```
I'll also be satisfied ;-)
Thanks again and sorry for being (a little bit) picky ;-)